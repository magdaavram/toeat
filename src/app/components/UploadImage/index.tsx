import React from 'react';
import styled from 'styled-components';
import ImageUploader from 'react-images-upload';

interface IProps {
  singleImage: boolean;
  withPreview: boolean;
  maxFileSize: number;
  label: string;
  fileSizeError: string;
  fileTypeError: string;
  imgExtension: string[];
  withIcon: boolean;
}

const containerStyle = {
  width: '300px',
  height: '300px',
  margin: 0,
  padding: '36px 21px 21px 21px',
  backgroundColor: 'var(--color--light-brown)',
  borderRadius: '9px',
  justifyContent: 'flex-start',
};

const buttonStyle = {
  width: '60px',
  height: '60px',
  fontFamily: 'Suisse SemiBold',
  fontSize: 'var(--font-size--xlarge)',
  color: 'var(--color--beige)',
  padding: '0',
  borderRadius: '50%',
  margin: '18px 0',
};

const labelStyle = {
  fontFamily: 'Suisse Bold',
  fontSize: 'var(--font-size--small)',
  color: 'var(--color--dark-purple)',
  marginTop: '9px',
};

const errorStyle = {
  fontFamily: 'Roboto Medium',
  fontSize: 'var(--font-size--small)',
  color: 'var(--color--burgundy)',
  textAlign: 'center',
};

const UploadImage = (props: IProps) => {
  const {
    singleImage,
    imgExtension,
    withPreview,
    maxFileSize,
    label,
    fileSizeError,
    fileTypeError,
    withIcon,
  } = props;

  return (
    <ImageUploader
      fileContainerStyle={containerStyle}
      singleImage={singleImage}
      imgExtension={imgExtension}
      withPreview={withPreview}
      buttonText={'+'}
      maxFileSize={maxFileSize}
      label={label}
      labelStyles={labelStyle}
      fileSizeError={fileSizeError}
      fileTypeError={fileTypeError}
      errorStyle={errorStyle}
      buttonStyles={buttonStyle}
      withIcon={withIcon}
    />
  );
};

export default UploadImage;
