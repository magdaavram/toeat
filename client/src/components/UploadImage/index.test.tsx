import React from 'react';
import { mount } from 'enzyme';
import UploadImage from './index';


describe('<UploadImage />', () => {
  let wrapper;
  const onChange = jest.fn();
  const props = {
    singleImage: true,
    withPreview: true,
    maxFileSize: 200,
    label: 'Label',
    fileSizeError: 'File size error',
    fileTypeError: 'File type error',
    imgExtension: ['.png', '.jpg'],
    withIcon: true,
    onChange,
  };

  beforeEach(() => {
    wrapper = mount(<UploadImage {...props} />);
  });

  afterEach(() => {
    wrapper = null;
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
