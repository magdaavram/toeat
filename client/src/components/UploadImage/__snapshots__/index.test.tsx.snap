// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UploadImage /> should render correctly 1`] = `
<UploadImage
  fileSizeError="File size error"
  fileTypeError="File type error"
  imgExtension={
    Array [
      ".png",
      ".jpg",
    ]
  }
  label="Label"
  maxFileSize={200}
  onChange={[MockFunction]}
  singleImage={true}
  withIcon={true}
  withPreview={true}
>
  <ReactImageUploadComponent
    accept="image/*"
    buttonClassName=""
    buttonStyles={
      Object {
        "borderRadius": "50%",
        "color": "var(--color--beige)",
        "fontFamily": "Roboto Regular",
        "fontSize": "var(--font-size--xxlarge)",
        "height": "60px",
        "margin": "18px 0",
        "padding": "0",
        "width": "60px",
      }
    }
    buttonText="+"
    buttonType="button"
    className=""
    defaultImages={Array []}
    errorClass=""
    errorStyle={
      Object {
        "color": "var(--color--burgundy)",
        "fontFamily": "Roboto Medium",
        "fontSize": "var(--font-size--small)",
        "textAlign": "center",
      }
    }
    fileContainerStyle={
      Object {
        "backgroundColor": "var(--color--light-brown)",
        "borderRadius": "9px",
        "height": "300px",
        "justifyContent": "flex-start",
        "margin": 0,
        "padding": "36px 21px 21px 21px",
        "width": "300px",
      }
    }
    fileSizeError="File size error"
    fileTypeError="File type error"
    imgExtension={
      Array [
        ".png",
        ".jpg",
      ]
    }
    label="Label"
    labelClass=""
    labelStyles={
      Object {
        "color": "var(--color--dark-purple)",
        "fontFamily": "Suisse Bold",
        "fontSize": "var(--font-size--small)",
        "marginTop": "9px",
      }
    }
    maxFileSize={200}
    name=""
    onChange={[Function]}
    singleImage={true}
    style={Object {}}
    withIcon={true}
    withLabel={true}
    withPreview={true}
  >
    <div
      className="fileUploader "
      style={Object {}}
    >
      <div
        className="fileContainer"
        style={
          Object {
            "backgroundColor": "var(--color--light-brown)",
            "borderRadius": "9px",
            "height": "300px",
            "justifyContent": "flex-start",
            "margin": 0,
            "padding": "36px 21px 21px 21px",
            "width": "300px",
          }
        }
      >
        <img
          alt="Upload Icon"
          className="uploadIcon"
          src="UploadIcon.svg"
        />
        <p
          className=""
          style={
            Object {
              "color": "var(--color--dark-purple)",
              "fontFamily": "Suisse Bold",
              "fontSize": "var(--font-size--small)",
              "marginTop": "9px",
            }
          }
        >
          Label
        </p>
        <div
          className="errorsContainer"
        />
        <button
          className="chooseFileButton "
          onClick={[Function]}
          style={
            Object {
              "borderRadius": "50%",
              "color": "var(--color--beige)",
              "fontFamily": "Roboto Regular",
              "fontSize": "var(--font-size--xxlarge)",
              "height": "60px",
              "margin": "18px 0",
              "padding": "0",
              "width": "60px",
            }
          }
          type="button"
        >
          +
        </button>
        <input
          accept="image/*"
          multiple={false}
          name=""
          onChange={[Function]}
          onClick={[Function]}
          type="file"
        />
        <div
          className="uploadPicturesWrapper"
        >
          <FlipMovePropConverter
            delay={0}
            disableAllAnimations={false}
            duration={350}
            easing="ease-in-out"
            enterAnimation="fade"
            getPosition={[Function]}
            leaveAnimation="fade"
            maintainContainerHeight={false}
            staggerDelayBy={0}
            staggerDurationBy={0}
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexWrap": "wrap",
                "justifyContent": "center",
                "width": "100%",
              }
            }
            typeName="div"
            verticalAlignment="top"
          >
            <FlipMove
              delay={0}
              delegated={
                Object {
                  "style": Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexWrap": "wrap",
                    "justifyContent": "center",
                    "position": "relative",
                    "width": "100%",
                  },
                }
              }
              disableAllAnimations={false}
              duration={350}
              easing="ease-in-out"
              enterAnimation={
                Object {
                  "from": Object {
                    "opacity": "0",
                  },
                  "to": Object {
                    "opacity": "",
                  },
                }
              }
              getPosition={[Function]}
              leaveAnimation={
                Object {
                  "from": Object {
                    "opacity": "1",
                  },
                  "to": Object {
                    "opacity": "0",
                  },
                }
              }
              maintainContainerHeight={false}
              staggerDelayBy={0}
              staggerDurationBy={0}
              typeName="div"
              verticalAlignment="top"
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexWrap": "wrap",
                    "justifyContent": "center",
                    "position": "relative",
                    "width": "100%",
                  }
                }
              />
            </FlipMove>
          </FlipMovePropConverter>
        </div>
      </div>
    </div>
  </ReactImageUploadComponent>
</UploadImage>
`;
